### DISCUSSION TITLE
std::set_new_handler and no exceptions

### DISCUSSION THREAD

**Discussion by stephenberry** [stephenberry/glaze#976] [Ideas]
Does [std::set_new_handler](https://en.cppreference.com/w/cpp/memory/new/set_new_handler) work if `-fno-exceptions` is set?

I have never experienced an allocation issues with the standard, because I think this typically only occurs when the system is running out of memory. I'd like to know the best way to handle these allocation errors in the context of `-fno-exceptions`.

**Comment by arturbac** [comment]
It depends on its internal code generated in stl.
If it is fully unrolled against any external exceptions handling symbols it will work and provide exception handling even when linked to -fno-exception build code.
But if it references any exception handling symbols provided at end of linking stage (just guessing) like any SEH external function symbols on windows that will be not available when not using exceptions, You will get unresolved symbols when linking.
Just check if it links, then it will work IMO.
Analogy You can link C library that is build without exceptions to C++ exceptions handling library exporting C symbols of functions and everything will link and work.
To the point when exception will be throw from C symbol in c+ generated code, then it will just not be forwarded further in C code.

**Reply by stephenberry** [reply]
Thanks for your insights, this is helpful and I'll do some experiments.
